src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.CalculateBalance;
+import core.basesyntax.service.FileReader;
+import core.basesyntax.service.ReportGenerator;
+import core.basesyntax.service.TransactionParser;
+import core.basesyntax.service.WriterFile;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private String report = "";
+    private FileReader fileReader = new FileReaderImpl();
+    private WriterFile fileWriter = new WriterFileImpl();
+    private CalculateBalance calculateBalance = new CalculateBalanceImpl();
+    private TransactionParser transactionParser = new TransactionParserImpl();
+
+    @Override
+    public String makeReport() {
+
+        for (String element: Storage.result.keySet()) {
+            report += (element + ", " + Storage.result.get(element) + System.lineSeparator());
+        }
+        return report;
+
+    }
```

Comment:
1. There is no need to have report as class variable. It may produce errors with consecutive method calls
2. Please use constructor to pass service we rely on. Otherwise code will be hard to maintain and test
3. Concatenation of String in cycle is a bad practice and wastes a lot of memory
```suggestion
public class ReportGeneratorImpl implements ReportGenerator {
    private final FileReader fileReader;
    private final WriterFile fileWrite;
    private final CalculateBalance calculateBalance;
    private final TransactionParser transactionParser;

    //add all-args constructor here

    @Override
    public String makeReport() {
        StringBuilder report = new StringBuilder();
        for (String element: Storage.result.keySet()) {
            report.append(element)
                .append(", ")
                .append(Storage.result.get(element))
                .append(System.lineSeparator());
        }
        return report.toString();
    }
```